main.tf

```diff
@@ -23,16 +23,18 @@ module "batch_subnet" {
 module "batch_queue" {
   source                = "./terraform/modules/swipe-sfn-batch-queue"
   app_name              = var.app_name
-  mock                  = var.mock
   batch_ssh_key_pair_id = length(aws_key_pair.swipe_batch) > 0 ? aws_key_pair.swipe_batch[0].id : ""
   network_info = length(module.batch_subnet) == 0 ? var.network_info : {
     vpc_id           = module.batch_subnet[0].vpc_id
     batch_subnet_ids = module.batch_subnet[0].batch_subnet_ids
   }
   ami_id                   = var.batch_ami_id
+  ami_ssm_parameter        = var.ami_ssm_parameter
```

Comment:
This is a replacement for the `mock` variable -- we can now pass in the path to an SSM parameter to read an overridden AMI id from.

Reply:
I really don't like `mock` being a simple boolean because:
- We take the shortcut of sending in a `mock` var when we could instead be exposing useful functionality
- We will likely have *different* mocking scenarios! I'm working on getting these tests to work with moto and localstack, and the two different mocking utilities require slightly different configurations.